packages/front-end/pages/datasources/[did].tsx

```diff
@@ -260,123 +207,36 @@ mixpanel.init('YOUR PROJECT TOKEN', {
-                          )}
-                        </div>
-                      </div>
-                    </div>
-                    <Code
-                      language="sql"
-                      theme="light"
-                      code={e.query}
-                      containerClassName="mb-0"
-                      expandable={true}
-                    />
-                  </div>
-                ))}
-              </div>
-
-              {/* region Jupyter Notebook */}
-              <div className="mb-4">
-                <div className="d-flex justify-content-between align-items-center">
-                  <div className="">
-                    <h3>Jupyter Notebook Query Runner</h3>
-                  </div>
-
-                  <div className="">
-                    <button
-                      className="btn btn-outline-primary font-weight-bold"
-                      onClick={() => {
-                        setUiMode("edit");
-                        setEditingResource("jupyter_notebook");
-                      }}
-                    >
-                      {d.settings.notebookRunQuery ? (
-                        <>
-                          <FaPencilAlt className="mr-1" /> Edit
-                        </>
-                      ) : (
-                        <>
-                          <FaPlus className="mr-1" /> Add
-                        </>
-                      )}
-                    </button>
-                  </div>
-                </div>
-                <p>
-                  Tell us how to query this data source from within a Jupyter
-                  notebook environment.
-                </p>
-                {d.settings?.notebookRunQuery ? (
-                  <Code
-                    theme="light"
-                    code={d.settings.notebookRunQuery}
-                    language="python"
-                    expandable={true}
-                  />
-                ) : (
-                  <div className="alert alert-info">
-                    Used when exporting experiment results to a Jupyter notebook
-                  </div>
-                )}
+              <div className="my-5">
+                <ExperimentAssignmentQueries
+                  dataSource={d}
+                  onSave={updateDataSource}
+                  onCancel={() => undefined}
+                />
               </div>
 
-              {d &&
-              uiMode === "edit" &&
-              editingResource === "jupyter_notebook" ? (
-                <EditJupyterNotebookQueryRunner
-                  onSave={updateDataSource}
-                  onCancel={cancelUpdateDataSource}
+              <div className="my-5">
+                <DataSourceJupyterNotebookQuery
                   dataSource={d}
+                  onSave={updateDataSource}
+                  onCancel={() => undefined}
                 />
-              ) : null}
-
-              {/* endregion Jupyter Notebook */}
+              </div>
```

Comment:
The Jupyter query section was reworked to have it manage its own editing state like the other sections.